opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/otlp_encoder.rs

```diff
@@ -0,0 +1,411 @@
+                        let hex = hex::encode(&log.span_id);
+                        BondWriter::write_string(&mut buffer, &hex);
+                    }
+                }
+                "env_dt_traceFlags" => {
+                    if log.flags != 0 {
+                        BondWriter::write_int32(&mut buffer, log.flags as i32);
+                    }
+                }
+                "name" => {
+                    if !log.event_name.is_empty() {
+                        BondWriter::write_string(&mut buffer, &log.event_name);
+                    }
+                }
+                "SeverityNumber" => BondWriter::write_int32(&mut buffer, log.severity_number),
+                "SeverityText" => {
+                    if !log.severity_text.is_empty() {
+                        BondWriter::write_string(&mut buffer, &log.severity_text);
+                    }
+                }
+                "body" => {
+                    // TODO - handle all types of body values
+                    // For now, we only handle string values
+                    if let Some(body) = &log.body {
+                        if let Some(Value::StringValue(s)) = &body.value {
+                            BondWriter::write_string(&mut buffer, s);
+                        }
+                    }
+                }
+                _ => {
+                    // Handle dynamic attributes
+                    if let Some(attr) = log.attributes.iter().find(|a| a.key == field.name) {
+                        self.write_attribute_value(&mut buffer, attr, field.type_id);
+                    }
+                }
+            }
+        }
+
+        buffer
+    }
+
+    /// Format timestamp from nanoseconds
+    fn format_timestamp(nanos: u64) -> String {
+        let secs = (nanos / 1_000_000_000) as i64;
+        let nsec = (nanos % 1_000_000_000) as u32;
+        Utc.timestamp_opt(secs, nsec)
+            .single()
+            .unwrap_or_else(|| Utc.timestamp_opt(0, 0).single().unwrap())
+            .to_rfc3339()
+    }
+
+    /// Write attribute value based on its type
+    fn write_attribute_value(
+        &self,
+        buffer: &mut Vec<u8>,
+        attr: &opentelemetry_proto::tonic::common::v1::KeyValue,
+        expected_type: u8,
+    ) {
+        if let Some(val) = &attr.value {
+            match (&val.value, expected_type) {
+                (Some(Value::StringValue(s)), t) if t == BondDataType::BT_STRING as u8 => {
+                    BondWriter::write_string(buffer, s)
+                }
+                (Some(Value::StringValue(s)), t) if t == BondDataType::BT_WSTRING as u8 => {
+                    BondWriter::write_wstring(buffer, s)
+                }
+                (Some(Value::IntValue(i)), t) if t == BondDataType::BT_INT32 as u8 => {
+                    BondWriter::write_int32(buffer, *i as i32)
+                }
+                (Some(Value::DoubleValue(d)), t) if t == BondDataType::BT_FLOAT as u8 => {
```

Comment:
Would be marking types as `float`? I think we would only use `double`, right?

Reply:
I just followed C++ implementation for now - it uses BT_FLOAT for double/float. But agreed, this is not right. OTLP supports only double, and better to encode it as double. Have added a TODO for this, and also created an issue to track this.